### PR TITLE
dsymbolsem.d: replace loops with foreachDsymbol()

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

File: src/dmd/dsymbolsem.d [hunk 1]
```diff
@@ -2205,16 +2200,8 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
...
-
-        // https://issues.dlang.org/show_bug.cgi?id=9701
-        if (em.ed.isAnonymous())
-        {
-            if (em.userAttribDecl)
-                em.userAttribDecl.userAttribDecl = em.ed.userAttribDecl;
-            else
-                em.userAttribDecl = em.ed.userAttribDecl;
-        }
+        em.storage_class = STC.manifest;
```
Comment (thewilsonator) [inline src/dmd/dsymbolsem.d]: this loses an `|` is that intended?

File: src/dmd/dsymbolsem.d [hunk 2]
```diff
@@ -2283,7 +2271,7 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
 
                 if (!em.ed.isAnonymous())
                 {
-                    e = e.castTo(sc, em.ed.type.addMod(e.type.mod)); // https://issues.dlang.org/show_bug.cgi?id=12385
+                    e = e.castTo(sc, em.ed.type);
```
Comment (thewilsonator) [inline src/dmd/dsymbolsem.d]: Does this bug no longer apply?

File: src/dmd/dsymbolsem.d [hunk 3]
```diff
@@ -2328,27 +2316,25 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
...
                 emprev.dsymbolSemantic(emprev._scope); // resolve it
             if (emprev.errors)
                 return errorReturn();
 
             Expression eprev = emprev.value;
-            // .toHeadMutable() due to https://issues.dlang.org/show_bug.cgi?id=18645
-            Type tprev = eprev.type.toHeadMutable().equals(em.ed.type.toHeadMutable())
-                ? em.ed.memtype
-                : eprev.type;
+            Type tprev = eprev.type.equals(em.ed.type) ? em.ed.memtype : eprev.type;
```
Comment (thewilsonator) [inline src/dmd/dsymbolsem.d]: Does this bug no longer apply?

Comment (WalterBright) [reply src/dmd/dsymbolsem.d]: Sorry, bad merge on my part. Fixed.

**Review by thewilsonator — APPROVED** [review]
LGTM.

**Pull request merged**